tls/doc/tls_introduction.md

```diff
@@ -0,0 +1,263 @@
+
+The OSI model classifies the typical building blocks of technical communication systems into seven layers.
+
+| Layer | Name               | Example         | TLS Components                              |
+|:-----:|--------------------|-----------------|---------------------------------------------|
+|   7   | Application layer  | HTTP            |                                             |
+|   6   | Presentation layer | **TLS**         | Encryption/Decryption, Data Compression     |
+|   5   | Session layer      | **TLS**         | Handshake, Session Management, Key Exchange |
+|   4   | Transport layer    | TCP             |                                             |
+|   3   | Network layer      | IP              |                                             |
+|   2   | Data link layer    | MAC             |                                             |
+|   1   | Physical layer     | Network adapter |                                             |
+
+The most obvious one is the physical layer. Here you find the hardware and the physical communication channel like an optical fiber. The data link layer provides a point-to-point connection between two nodes. The network layer allows you to transfer data to other nodes beyond that direct neighbor. It’s the job of the transport layer to make sure that the data arrives intact on the other side of the communication. Especially if that data is too large to fit into a single transfer unit and needs to be split at the source, correctly reassembled at the sink.
+
+The session layer is responsible for managing the connection state between communication partners. TLS has a session concept — so a part of TLS lives in this layer. The presentation layer is a translator between what the application expects and what the lower layers provide. In the case of TLS, this means encryption and decryption of application data.
+
+The table above shows an example of an HTTPS connection. While you can see that the association between protocols and most layers is clear, TLS affects parts of both the session and presentation layer.
+
+**TLS Session Layer (Layer 5) responsibilities:**
+
+- Managing the TLS handshake process
+- Establishing and maintaining secure sessions
+- Negotiating cipher suites and protocol versions
+- Handling session resumption and renegotiation
+- Managing connection state between communication partners
+
+**TLS Presentation Layer (Layer 6) responsibilities:**
+
+- Encrypting outgoing application data
+- Decrypting incoming encrypted data
+- Data compression and decompression (when enabled)
+- Converting between different data formats as needed
+
+This layered approach means that when Alice sends an HTTPS request to Bob's server, the application data first gets encrypted by TLS's presentation layer functions. While the session layer functions ensure the secure connection is properly maintained throughout the communication.
```

Comment:
Not sure how to re-word to maintain content, but `While the session layer functions ensure the secure connection is properly maintained throughout the communication.` isn't a complete sentence currently.

"Supporting this, is the session layer which ensures the secure connection is maintained throughout the communication"